policy-controller/k8s/index/src/inbound/index.rs

```diff
@@ -1547,32 +1568,33 @@ impl PolicyIndex {
 
         let authorizations = policy.default_authzs(config);
 
-        let http_routes = config.default_inbound_http_routes(probe_paths);
-
         InboundServer {
-            reference: ServerRef::Default(policy.as_str()),
             protocol,
             authorizations,
-            http_routes,
+            reference: ServerRef::Default(policy.as_str()),
+            http_routes:
+                <ClusterInfo as DefaultInboundRoutes<HttpRouteMatch>>::default_inbound_routes(
+                    config,
+                    probe_paths,
+                ),
         }
     }
 
     fn inbound_server<'p>(
         &self,
-        name: String,
+        name: &str,
```

Comment:
what's the purpose this change? `String` seems more honest than `&str` since we're going to `to_string()` it anyway.

Reply:
It saves pushes the clone to a single place (inside the function itself) instead of forcing callers to provide an owned value to begin with

![Screenshot 2024-06-10 at 1 06 13 PM](https://github.com/linkerd/linkerd2/assets/61921871/2a9d64e6-e949-4187-ac39-42c6b1abdcd6)